### ISSUE TITLE
A11y: ToC Block links are not marked as ToC links 

### ISSUE THREAD

**Issue by Tishasoumya-02** [plone/volto#6082]
**Problem:** Every ToC-Link are not labeled as such. Whenever a screenreader is used, it doesn't tell the user that the link is a ToC-Link. The link seems to be linking to another page, but that is not the case. It's always an anchor link. This might confuse the user.

**Proposed Solution:** Mark the links as table-of-contents-links with alt-tags or ARIA-labels. A screenreader should read "Title-of-the-link, table-of-content-link" or something similar.

**Steps to reproduce:**

* Create Table-of-Contents block in page and populate it.
* Inspect Table-of-Contents-Links or use a screenreader
* Link is not shown as ToC-Link, no aria label, no alt tag

**Expected result:**

* The link should be marked as Table-of-Contents-Link

**WCAG - 2.1:**

* https://www.w3.org/TR/WCAG21/#label-in-name

**BIK BITV-Test (Web):**

* 9.2.5.3: https://ergebnis.bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-2-5-3-sichtbare-beschriftung-teil-des-zugaenglichen-namens

**Comment by ichim-david** [comment]
I left a review on the pull request https://github.com/plone/volto/pull/6084#pullrequestreview-2105250544
From my point of view this proposed solution does more harm than good and although we can improve the toc accesibility by ensuring we have good markup.

**Comment by sabrina-bongiovanni** [comment]
If the other PR is going to solve this issue, do we still need to change the markup on this one?
